### PR TITLE
Add `min` and `max` operators

### PR DESCRIPTION
**Pull request by twyatt**
Useful for computing the domain of data when the data contains multiple values that may influence the min or max.

For example, `max` is used in [Line plot with several groups](https://www.d3-graph-gallery.com/graph/line_several_group.html) for computing the y-axis domain:

Screen Shot 2021-12-31 at 12 55 25 PM

They were also used in https://github.com/JuulLabs/sensortag/pull/48.

### CODE REVIEW

File: scale/src/commonMain/kotlin/Max.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.juul.krayon.scale
+
+public inline fun <I, O : Comparable<O>> List<I>.max(
```
Comment (twyatt) [inline scale/src/commonMain/kotlin/Max.kt:3]: To be more general, should this be changed to?:

```suggestion
public inline fun <I, O : Comparable<O>> Collection<I>.max(
```

Comment (sdonn3) [reply scale/src/commonMain/kotlin/Max.kt:3]: Seems pretty reasonable

**Review by sdonn3 — APPROVED** [review]

**Review by cedrickcooke — APPROVED** [review]
We'll probably want to deprecate these in a later Kotlin release, once they finish their changes to the built-in min/max functionality. But love having these for now.

File: scale/src/commonMain/kotlin/Max.kt (hunk 1 above)
Comment (twyatt) [reply scale/src/commonMain/kotlin/Max.kt:3]: #75

**Pull request merged**